### DISCUSSION TITLE
Thoughts on using NextJS/Vercel as an Auth Server?

### DISCUSSION THREAD

**Discussion by michaelmerrill** [vercel/next.js#19023] [Help] [answered]
Hi All,

I'm wondering if NextJs and Vercel are good tools to use for developing and deploying an authorization server. I'd rather use them than the more common express/ejs stack. Any thoughts on this?

Here are the example routes:
```
GET    accounts.example.com/login
GET    accounts.example.com/authorize
POST   accounts.example.com/api/login
GET    accounts.example.com/api/logout
POST   accounts.example.com/api/token
```

**Comment by lfades** [comment, marked as answer]
Yes, you should be able to do that with no issues with Next.js and Vercel, we already have many examples doing authentication and this is a very good use case for API Routes 👍 

> Btw if your API route doesn't start with `/api` you can use Next.js rewrites to change it 👀
